src/runtime_src/core/common/api/aie/xrt_graph.cpp

```diff
@@ -124,61 +125,48 @@ namespace xrt::aie {
 class profiling_impl
 {
 private:
-  std::shared_ptr<xrt_core::device> device;
-  int profiling_hdl;
+  std::unique_ptr<xrt_core::profile_handle> m_profile_handle{nullptr};
 
 public:
-  using handle = int;
-  static constexpr handle invalid_handle = -1;
+  static constexpr int invalid_handle = -1;
 
-  explicit profiling_impl(std::shared_ptr<xrt_core::device> dev)
-    : device(std::move(dev)),
-      profiling_hdl(invalid_handle)
+  profiling_impl(std::shared_ptr<xrt_core::device> device)
+    : m_profile_handle{device->open_profile_handle()}
   {}
 
+  profiling_impl(const xrt::hw_context& hwctx)
+  {
+    auto hwctx_handle = static_cast<xrt_core::hwctx_handle*>(hwctx);
+    m_profile_handle = hwctx_handle->open_profile_handle();
+  }
+
   ~profiling_impl()
   {
-    try {
-      if (profiling_hdl != invalid_handle)
-        device->stop_profiling(profiling_hdl);
-    }
-    catch(...) {
-      // do nothing
-    }
+    stop_profiling();
   }
 
   profiling_impl() = delete;
   profiling_impl(const profiling_impl&) = delete;
   profiling_impl(profiling_impl&&) = delete;
   profiling_impl& operator=(const profiling_impl&) = delete;
   profiling_impl& operator=(profiling_impl&&) = delete;
-  
 
-  handle
+  int
   start_profiling(int option, const std::string& port1_name, const std::string& port2_name, uint32_t value)
   {
-    profiling_hdl = device->start_profiling(option, port1_name.c_str(), port2_name.c_str(), value);
-    return profiling_hdl;
+    return m_profile_handle->start_profiling(option, port1_name.c_str(), port2_name.c_str(), value);
```

Comment:
as class name already has profiling name in it, just call function names as start/read/stop.